doc/changelog/05-tactic-language/9122-genarg_cleanup.rst

```diff
@@ -0,0 +1,10 @@
+- **Changed:**
+  Some duplicated grammar binders in tactic notations have been
+  removed, for example in ``Tactic Notation "foo" ref(x) := idtac.``
+  you now must use ``reference(x)``, etc.... The full list is:
+  + `integer` -> `int`
```

Comment:
Does `int` even work with `Tactic Notation`?  I was under the impression that only `integer` worked.

Reply:
Also, it seems wrong to change `integer` to `int` but `ref` to `reference`.  All the Ltac code in the wild I'm aware of uses `integer` and `reference`, I think we should standardize on that

Reply:
Actually there are some extra compatibility notations registered in `extraargs.mlg` which indeed is making this work; I'm unsure what the relation is here, @ppedrot ? I've only updated the "core" ones.

Reply:
> Does `int` even work with `Tactic Notation`? I was under the impression that only `integer` worked.

At least today both work, one from the core arg mechanism and the other from the compat layer I mentioned. I'm ok switching everything to `integer` if that's the preferred solution, but I'd like to get rid of duplicates. I got some users in the ML world that indeed would like the syntax to be more canonical.

Reply:
For the ones accessible via `Tactic Notation`, let's standardize on the ones [currently document](https://coq.inria.fr/refman/user-extensions/syntax-extensions.html#tactic-notations), that is:
`ident`, `simple_intropattern`, `reference`, `hyp`, `hyp_list`, `ne_hyp_list`, `constr`, `uconstr`, `constr_list`, `ne_constr_list`, `integer`, `integer_list`, `ne_integer_list`, `int_or_var`, `int_or_var_list`, `ne_int_or_var_list`, `tactic`, `tactic0`, `tactic1`, `tactic2`, `tactic3`, `tactic4`, `tactic5`